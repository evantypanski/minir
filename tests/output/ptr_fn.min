func main() -> none {
  let i = 100;
  debug(i);
  add_one_sanity_check(i)
  debug(i);
  let ptr = alloc(int);
  *ptr = 5
  debug(*ptr);
  add_one(ptr)
  debug(*ptr);
  add_one(identity(ptr))
  debug(*ptr);
  ret
}

func add_one_sanity_check(i: int) -> none {
  i = i + 1
  ret
}

func add_one(i: *int) -> none {
  *i = *i + 1
  ret
}

func identity(i: *int) -> *int {
  ret i
}
