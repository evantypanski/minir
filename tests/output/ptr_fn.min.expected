100
100
5
6
7
